value/src/test/java/com/google/auto/value/processor/ExtensionTest.java

```diff
@@ -120,12 +121,49 @@ public void testExtensionConsumesProperties() throws Exception {
     );
     assertAbout(javaSource())
         .that(javaFileObject)
-        .processedWith(
-            new AutoValueProcessor(ImmutableList.<AutoValueExtension>of(new FooExtension())))
+        .processedWith(new AutoValueProcessor(ImmutableList.of(new FooExtension())))
         .compilesWithoutError()
         .and().generatesSources(expectedExtensionOutput);
   }
 
+  public void testDoesntRaiseWarningOnConsumedProperties() {
+    JavaFileObject impl = JavaFileObjects.forSourceLines("foo.bar.Baz",
+        "package foo.bar;",
+        "import com.google.auto.value.AutoValue;",
+        "@AutoValue public abstract class Baz {",
+        "  abstract String foo();",
+        "  abstract String dizzle();",
+        "}");
+    assertAbout(javaSource())
+        .that(impl)
+        .processedWith(new AutoValueProcessor(ImmutableList.of(new FooExtension())))
+        .compilesWithoutError()
+        // expected warnings: @AutoValue and @Generated annotations not consumed
+        .withWarningCount(2);
```

Comment:
I've had success passing `.withCompilerOptions("-Xlint:-processing")` to turn off the warnings about `@AutoValue` and `@Generated` not being consumed by any processor. However, when I do that in this test, it passes without any other changes to the code. I think we need a test that fails before the rest of the modifications in this PR are made and passes afterwards.